### PR TITLE
PluginBase::getResources() returns associative array

### PR DESCRIPTION
**Pull request by SOF3**
### API changes

`PluginBase::getResources()` now returns the result in an associative array of strings.

Previously, it was a plain linear array containing instances of `SplFileInfo`.

Now, it has string keys, which is the path of the resource file relative to the resources folder.

## Backwards compatibility

May affect plugins that expect getResources() to return numeric arrays. However, the key type has never been clearly documented, so it is not considered an API break.

[A search on Poggit](https://poggit.pmmp.io/grepPlugins/2018-05-20-439be3a5b802b571.html) shows that no plugins are relying on the numeric nature of the keys.

## Tests
Tested.

### CODE REVIEW

File: src/pocketmine/plugin/PluginBase.php [hunk 1]
```diff
@@ -229,7 +229,7 @@ public function getResources() : array{
 		$resources = [];
 		if(is_dir($this->file . "resources/")){
 			foreach(new \RecursiveIteratorIterator(new \RecursiveDirectoryIterator($this->file . "resources/")) as $resource){
-				$resources[] = $resource;
+				$resources[substr($resource, strlen($this->file . "resources/"))] = $resource;
```
Comment (dktapps) [inline src/pocketmine/plugin/PluginBase.php]: this probably won't work as you expect on Windows (different directory separator), please check it.

**Comment by dktapps** [comment]
Is this ready to merge?

**Comment by SOF3** [comment]
Test result with this structure:

```
resources/
├── resources/a.b
├── resources/c/
    ├──resources/c.d
```

Output as a folder plugin loaded with DevTools:

```
array(2) {
  ["a.b"]=>
  object(SplFileInfo)#9829 (2) {
    ["pathName":"SplFileInfo":private]=>
    string(62) "C:\sofe\proj\PocketMine-MP\plugins\testResources/resources\a.b"
    ["fileName":"SplFileInfo":private]=>
    string(3) "a.b"
  }
  ["c\d"]=>
  object(SplFileInfo)#9832 (2) {
    ["pathName":"SplFileInfo":private]=>
    string(62) "C:\sofe\proj\PocketMine-MP\plugins\testResources/resources\c\d"
    ["fileName":"SplFileInfo":private]=>
    string(1) "d"
  }
}
```

As a bundled phar plugin:
```
array(2) {
  ["a.b"]=>
  object(SplFileInfo)#9829 (2) {
    ["pathName":"SplFileInfo":private]=>
    string(74) "phar://C:\sofe\proj\PocketMine-MP\plugins\testResources.phar/resources\a.b"
    ["fileName":"SplFileInfo":private]=>
    string(3) "a.b"
  }
  ["c\d"]=>
  object(SplFileInfo)#9832 (2) {
    ["pathName":"SplFileInfo":private]=>
    string(74) "phar://C:\sofe\proj\PocketMine-MP\plugins\testResources.phar/resources\c\d"
    ["fileName":"SplFileInfo":private]=>
    string(1) "d"
  }
}
```

I recommend adding `str_replace(DIRECTORY_SEPARATOR, "/")` on all keys in order to produce consistent keys.

**Comment by dktapps** [comment]
there's an unrelated submodule change here

**Pull request merged**